CMake build fails on Arch Linux when building with zlib

When using cmake to build partio, the build fails with the following (snippit) of errors:
```
../../lib/libpartio.a(ZIP.o): In function `Partio::ZipFileReader::Get_File(std::string const&, bool)':
ZIP.cpp:(.text+0xd63): undefined reference to `inflateInit2_'
../../lib/libpartio.a(ZIP.o): In function `Partio::Gzip_In(std::string const&, std::_Ios_Openmode)':
ZIP.cpp:(.text+0x15c9): undefined reference to `inflateInit2_'
../../lib/libpartio.a(ZIP.o): In function `Partio::Gzip_Out(std::string const&, std::_Ios_Openmode)':
ZIP.cpp:(.text+0x1b08): undefined reference to `deflateInit2_'
../../lib/libpartio.a(ZIP.o): In function `Partio::ZipFileWriter::Add_File(std::string const&, bool)':
ZIP.cpp:(.text+0x1fcc): undefined reference to `deflateInit2_'
../../lib/libpartio.a(ZIP.o): In function `Partio::ZipStreambufCompress::overflow(int)':
ZIP.cpp:(.text._ZN6Partio20ZipStreambufCompress8overflowEi[_ZN6Partio20ZipStreambufCompress8overflowEi]+0x65): undefined reference to `deflate'
ZIP.cpp:(.text._ZN6Partio20ZipStreambufCompress8overflowEi[_ZN6Partio20ZipStreambufCompress8overflowEi]+0xcd): undefined reference to `crc32'
...
```

I believe this is caused by the partio library depending on zlib but the link order is incorrect:
```
/usr/bin/c++   -fPIC -O3 -DNDEBUG   -Wl,-O1,--sort-common,--as-needed,-z,relro CMakeFiles/partattr.dir/partattr.o  -o ../../bin/partattr -rdynamic -lz ../../lib/libpartio.a
```
should be
```
/usr/bin/c++   -fPIC -O3 -DNDEBUG   -Wl,-O1,--sort-common,--as-needed,-z,relro CMakeFiles/partattr.dir/partattr.o  -o ../../bin/partattr -rdynamic ../../lib/libpartio.a -lz
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -93,7 +93,7 @@ ENDIF(ZLIB_FOUND)
 
 ## Make modules able to see partio library
 # Setup environment variable to link partio
-SET( PARTIO_LIBRARIES ${ZLIB_LIBRARY} partio )
+SET( PARTIO_LIBRARIES partio ${ZLIB_LIBRARY} )
 # make it so partio can be found
 INCLUDE_DIRECTORIES( ${CMAKE_SOURCE_DIR}/src/lib )
 

